src/append/rolling_file/mod.rs

```diff
@@ -429,49 +465,146 @@ appenders:
         }
     }
 
+    #[derive(Debug)]
+    struct NopPrePolicy;
+
+    impl Policy for NopPrePolicy {
+        fn process(&self, _: &mut LogFile) -> anyhow::Result<()> {
+            Ok(())
+        }
+        fn is_pre_process(&self) -> bool {
+            true
+        }
+    }
+
     #[test]
-    fn append() {
-        let dir = tempfile::tempdir().unwrap();
-        let path = dir.path().join("append.log");
-        RollingFileAppender::builder()
-            .append(true)
-            .build(&path, Box::new(NopPolicy))
-            .unwrap();
-        assert!(path.exists());
-        File::create(&path).unwrap().write_all(b"hello").unwrap();
-
-        RollingFileAppender::builder()
-            .append(true)
-            .build(&path, Box::new(NopPolicy))
-            .unwrap();
-        let mut contents = vec![];
-        File::open(&path)
-            .unwrap()
-            .read_to_end(&mut contents)
-            .unwrap();
-        assert_eq!(contents, b"hello");
```

Comment:
this seems like a different test right? Seems like we should keep it if so

Reply:
This was combined with the test below. To hit the pre_process vs post_process writing sections. That said, let me add a check on the file contents.